README
NAME
    MooX::Role::Logger - Provide logging via Log::Any

VERSION
    version 0.004

SYNOPSIS
    In your modules:

        package MyModule;
        use Moose;
        with 'MooX::Role::Logger';

        sub run {
            my ($self) = @_;
            $self->cry;
        }

        sub cry {
            my ($self) = @_;
            $self->_logger->info("I'm sad");
        }

    In your application:

        use MyModule;
        use Log::Any::Adapter ('File', '/path/to/file.log');

        MyModule->run;

DESCRIPTION
    This role provides universal logging via Log::Any. The class using this
    role doesn't need to know or care about the details of log
    configuration, implementation or destination.

    Use it when you want your module to offer logging capabilities, but
    don't know who is going to use your module or what kind of logging they
    will implement. This role lets you do your part and leaves actual log
    setup and routing to someone else.

    The application that ultimately uses your module can then choose to
    direct log messages somewhere based on its own needs and configuration
    with Log::Any::Adapter.

    This role is based on Moo so it should work with either Moo or Moose
    based classes.

USAGE
  Testing
    Testing with Log::Any is pretty easy, thanks to Log::Any::Test. Just
    load that before Log::Any loads and your log messages get sent to a test
    adapter that includes testing methods:

        use Test::More 0.96;
        use Log::Any::Test;
        use Log::Any qw/$log/;

        use lib 't/lib';
        use MyModule;

        MyModule->new->cry;
        $log->contains_ok( qr/I'm sad/, "got log message" );

        done_testing;

  Customizing
    If you have a whole set of classes that should log with a single
    category, create your own role and set the "_build__logger_category"
    there:

        package MyLibrary::Role::Logger;
        use Moo::Role;
        with 'MooX::Role::Logger';

        sub _build__logger_category { "MyLibrary" }

    Then in your other classes, use your custom role:

        package MyLibrary::Foo;
        use Moo;
        with 'MyLibrary::Role::Logger'

METHODS
  _logger
    Returns a logging object. See Log::Any for a list of logging methods it
    accepts.

  _build__logger_category
    Override to set the category used for logging. Defaults to the class
    name of the object (which could be a subclass). You can override to lock
    it to a particular name:

        sub _build__logger_category { __PACKAGE__ }

SEE ALSO
    Since MooX::Role::Logger is universal, you have to use it with one of
    several Log::Any::Adapter classes:

    *   Log::Any::Adapter::File

    *   Log::Any::Adapter::Stderr

    *   Log::Any::Adapter::Stdout

    *   Log::Any::Adapter::ScreenColoredLevel

    *   Log::Any::Adapter::Dispatch

    *   Log::Any::Adapter::Syslog

    *   Log::Any::Adapter::Log4perl

    These other logging roles are specific to particular logging packages,
    rather than being universal:

    *   MooseX::LazyLogDispatch

    *   MooseX::Log::Log4perl

    *   MooseX::LogDispatch

    *   MooseX::Role::LogHandler

    *   MooseX::Role::Loggable (uses Log::Dispatchouli)

    *   Role::Log::Syslog::Fast

SUPPORT
  Bugs / Feature Requests
    Please report any bugs or feature requests through the issue tracker at
    <https://github.com/dagolden/MooX-Role-Logger/issues>. You will be
    notified automatically of any progress on your issue.

  Source Code
    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    <https://github.com/dagolden/MooX-Role-Logger>

      git clone https://github.com/dagolden/MooX-Role-Logger.git

AUTHOR
    David Golden <dagolden@cpan.org>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2013 by David Golden.

    This is free software, licensed under:

      The Apache License, Version 2.0, January 2004



Changes
Revision history for MooX-Role-Logger

0.004     2014-09-18 10:32:10-04:00 America/New_York

    [Changed]

    - Renamed to MooX::Role::Logger for clarify that it works with Moo or
      Moose.

    - MooseX::Role::Logger deprecated but kept as a wrapper for backwards
      compatibility.

0.003     2014-07-03 12:38:11-04:00 America/New_York

    [Changed]

    - Drop minimum required perl to 5.006

0.002     2014-01-24 16:50:08-05:00 America/New_York

    [Incompatible Change]

    - The role now provides private method '_logger' rather than
      public method 'logger' (thanks, mst)

0.001     2013-10-03 00:24:33 America/New_York

    - First release



lib/MooseX/Role/Logger.pm
use strict;
use warnings;

package MooseX::Role::Logger;
# ABSTRACT: Provide logging via Log::Any (DEPRECATED)
our $VERSION = '0.004'; # VERSION

use Moo::Role;
with 'MooX::Role::Logger';

1;


# vim: ts=4 sts=4 sw=4 et:

__END__

=pod

=encoding UTF-8

=head1 NAME

MooseX::Role::Logger - Provide logging via Log::Any (DEPRECATED)

=head1 VERSION

version 0.004

=head1 DESCRIPTION

L<MooseX::Role::Logger> has been renamed to L<MooX::Role::Logger> to clarify
that it works with both L<Moo> and L<Moose>.  This role just wraps that one and
is provided for backwards compatibility.

See L<MooX::Role::Logger> for usage details.

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2013 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.17;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Provide logging via Log::Any",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.17"
  },
  "DISTNAME" => "MooX-Role-Logger",
  "EXE_FILES" => [],
  "LICENSE" => "apache",
  "NAME" => "MooX::Role::Logger",
  "PREREQ_PM" => {
    "Log::Any" => 0,
    "Moo::Role" => 0,
    "Types::Standard" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Log::Any::Test" => 0,
    "Moo" => 0,
    "Test::FailWarnings" => 0,
    "Test::More" => "0.96",
    "lib" => 0
  },
  "VERSION" => "0.004",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Log::Any" => 0,
  "Log::Any::Test" => 0,
  "Moo" => 0,
  "Moo::Role" => 0,
  "Test::FailWarnings" => 0,
  "Test::More" => "0.96",
  "Types::Standard" => 0,
  "lib" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



